### DISCUSSION TITLE
Carreira Profissional, Back-End ou Front-End?

### DISCUSSION THREAD

**Discussion by furabio** [frontendbr/forum#353] [Discussão]
Estava me aventurando na internet e achei interessante trazer aqui para uma discussão saudável é claro, essa pergunta e a resposta dada pelo usuário **Jothaz** : 

> Ola galera irei cria um artigo falando sobre os temas apresentados no titulos , preciso de algumas opniões de vocês a respeito dessas duas profissoes.
> Qual seguir ? qual a vantagem de uma sobre a outra ? qual tem maior empregabilidade ?
> Poderiam me da essa força ? Deixem seu ponto de vista sobre: back-end e o front-end.
> Podem opnar com suas propiás palavras.

a resposta do usuário **Jothaz** foi a seguinte : 

> Acho que em primeiro existe uma confusão! O que vocês acham função de programador front-end é na verdade função de Webdesign. Ele é que define toda a acessecibilidade, usabilidade, lay-out a aparência do site. 
> 
> Programador front-end normalmente é o estagiário ruim de serviço que para evitar que ele faça um cagada libera-se para que ele trabalhe somente na parte web.
> 
> Imagine que eu tenha uma aplicação critica parada na empresa e tenho de enviar alguém para verificar o que esta acontecendo. Ai mando um programador front-end mas o promblema é na base. só que o programador front-end não sabe acessar a base e conferir. Então para resolver um problema tenho de destacar dois profissionais um programador front-end e um back-end? Já imaginou que bagunça que seria. Seria totalmente improdutivo.
> 
> Agora ser for um projeto como NodesJS (usando Mean stack) e o programador front-end form mestre e NodesJs, Exceprion, JavaScritp, AngluasJs, jQyery MVC e banco de dados NoSQL e OO. Ai pode ser. Mas hoje ainda não existem projetos voltados para um cenário assim. Mas no futuro pode ser. Mas mesmo você~e tem conhecer muito mais muito mesmo para prescindir do back-end.
> 
> Se você quer mesmo se profissionalizar e se destacar aprenda uma tecnologia back-end, do contrário nem vão te chamar para entrevistas e quando achar uma colocação vão lhe pagar um miséria. Eu prefiro um ótimo programador back-end do que um ótimo programador front-end. Afinal é muito mais fácil um programador back-end aprender a se virar com front-end do que o contrário.

O que acham, concordam ? discordam ? deixem suas opiniões e ponto de vista sobre o assunto.

**Comment by robsonbittencourt** [comment]
Meio difícil de ler o texto :)

Mas na minha opinião ele é muito extremista. Na real se for seguir a lógica dele, para resolver um problema no banco, você precisária de um DBA e não de um programador backend. 

Nos últimos tempos eu venho tentando me despreender dessa separação entre frontend e backend. Se alguém me pergunta qual é a minha função, respondo: Desenvolvedor de Software.

Pelo que acompanho aqui no fórum e em outros lugares, tem muita gente que trabalha em empresas que possuem essa separação de responsabilidades bem definidas. Mas nas empresas que trabalhei até agora, sempre tive que fazer os dois papéis.

Tenho mais experiência com backend (java), mas a tempo venho estudando Javascript/Node.js para ampliar meus conhecimentos e conseguir trabalhar em mais partes dos projetos.

**Comment by woliveiras** [comment]
> Programador front-end normalmente é o estagiário ruim de serviço que para evitar que ele faça um cagada libera-se para que ele trabalhe somente na parte web.

Posta o link dessa resposta ae, pois acho que vai dar uma treta boa lá! HHUAHUHUAHUAHUAU

Creio que o **Jothaz** não manja muito o que um Front Ender faz hoje em dia ou eu não entendi direito o que ele quis dizer.

>  Ele é que define toda a acessecibilidade, usabilidade, lay-out a aparência do site.

Eu nunca fiz isso e não manjo de Design.

Se me mandar desenhar uma tela, se prepare pra alguns risquinhos na tela com fundo branco.

Quem dera todo Back Ender manjasse bem de JavaScript, CSS avançado, Acessibilidade, Performance no Front End, Internationalization, etc, etc, etc... Creio eu, que ele só está subestimando a profissão por falta de conhecimento no que um Front Ender faz, de verdade.

Colocar um Back Ender, que **não gosta** de trabalhar com Client Side, fazendo Web Apps, só vai sair código feito com ódio no coração e de **difícil manutenção**.

Vai sair um pouco mais caro refazer todo código ruim quando a aplicação parar de escalar ou mesmo contratar um time de operações criticas (existem consultorias especializadas em desfazer a cagada dos outros) pra corrigir os erros de alguém sem experiência no Client.

No final ele só levantou o ponto de que **pra ele** um desenvolvedor só é bom quando é Full Stack.

Provavelmente o artigo que será escrito servirá para difundir a ideia dele de que o desenvolvedor deve ser Full Stack.

Não é tão fácil assim um Back Ender vir pro Front e fazer um código bonito tanto quanto não é fácil pra um Front Ender ir pro Back End e fazer um código bom.

Ambas profissões tem suas complexidades onde só se chega com um pouco de especialização e alguns anos de XP em uma área.

Full Stack, ambos podem ser e possuir suas especializações. Uma área não anula a outra. Só depende da pessoa querer ou não fazer isso.

> Se você quer mesmo se profissionalizar e se destacar aprenda uma tecnologia back-end, do contrário nem vão te chamar para entrevistas e quando achar uma colocação vão lhe pagar um miséria.

Fico com medo do que pessoas preconceituosas assim, escrevendo e instruindo por ai, podem fazer com a cabeça de um iniciante.

Mas, tentando responder as perguntas citadas com meu ponto de vista: 

**Qual seguir?**

Isso é bem mais pessoal do que decisão de mercado. A pessoa deve pesquisar sobre qual área lhe agrada mais e ir pra cima.

Claro que também tem o ponto do mercado, onde fica complicado escolher ser Front Ender onde não existe vaga de Front End, mas não é só por isso que alguém deve desistir da sua escolha. 

Depois que decidir qual área escolher ainda vai começar a treta de qual tecnologia escolher. Se for Back Ender, por exemplo, escolher entre Ruby, Python, Java, JS, C#, etc, por exemplo. Se Front Ender, vai se especializar no CSS ou no JS, por exemplo. Não é tão simples quanto tirar par ou impar.

**Qual vantagem de uma sobre a outra?**

Percebo que os Back Enders são mais respeitados como Engenheiros de Software do que os Front Enders. Percebo que, muitas empresas deixam o Front somente tomando decisões relacionadas a Telas do que Arquitetura de Software, apesar do cenário estar mudando com as Web Apps.

**Qual tem maior empregabilidade?**

[Um Dev Python ganha mais do que um Dev JavaScript](https://angel.co/salaries), porém Front End é uma profissão que está em alta no mercado. Muita vaga aqui (nas capitais), muita vaga pra fora do país e muita procura por bons profissionais, principalmente os que tem boa experiência com JS.

## 

Acho que essa confusão de posições em um projeto acontece por que pouca gente entende o que um Front Ender faz.

**Comment by woliveiras** [comment]
>  não vale a pena pagar um "fio de kenga" um salario mensal para "meter a mão na massa HTML, CSS, javaScript".

Lamentável;

http://www.devmedia.com.br/forum/carreira-profissional-back-end-ou-front-end/551193

**Comment by bernardodiasc** [comment]
"End" é uma parte da atividade/produto, "end" é o fim, de finalidade. O trabalho front end é para o front, para o browser, devices, usuários finais, etc. O trabalho back end é backstages, o server, banco de dados, sistema, infra, etc.

A confusão se dá pela falta de ampla experiência, ou falta de técnica mesmo. Isso pois é perfeitamente possível fazer projetos sem distinçao clara das responsabilidades, e nao tem nada de errado nisso, o fato é que existem formas melhores de se fazer sistemas, logo a falta de tecnica. 

Um dev pode ser "full stack" muito facilmente, isso não tem nada a ver com ser um bom ou experiente programador, nao tem nada a ver. Qualquer um pode dizer que é "full stack" se botar sozinho um site inteiro online. Full stack nao define qualidade, mas refere-se a quem compreende o stack inteiro do projeto, e nao se engane, existem milhares de stacks, ninguem sabe de tudo.

Vamos la, recapitulando: front end = browser, back end = server. As linguagens de programaçao e tecnologias (ferramentas) sao os meios, browser e server sao os fins. 🎉

**Comment by YuriBrunetto** [comment]
Eu tô sem palavras pra expressar o quão ignorante esse cara é. Claramente é um daqueles dev's que acham que o front-end é feito por pessoas que não sabem programar. Ele programa e se acha superior, por não saber estruturar um HTML e um CSS.

Realmente, lamentável.

**Comment by Kirmayrtomaz** [comment]
Pessoas que não conhecem da área, acham que frontend deveria ser webdesign. 
Na moral, também fico muito chateado com essa situação

Programador Front-end pensa em tantas coisas como SEO, performace, utilização de modularização de css com algum pre-processador, js e seus supersets, lints, frameworks.
APIS do HTML5, armazenamento offline...e tantas outras coisas ....
Lamentável

**Comment by expalmer** [comment]
Pra mim o cara está só zuando! Está errando as palavras de propósito...apenas para gerar alvoroço...e conseguiu kkk

**Comment by herberthenrique** [comment]
Bem, vamos lá
Eu tenho todo um passado de back-end que odiava mexer em html, css  (bem idiota eu sei) e acho que entendo o que acontece quando você responde um algo daquele jeito.

O que acontece com a resposta do jovem rapaz lá é apenas um visão equivocada do como o mundo é, sabem aquela velha história de que "o que eu faço é melhor"? Isso causa milhões de coisas ruins na nossa area, como a já cansativa guerra de linguagens (boring).

Hoje, ao conhecer alguém novo na area eu sempre digo para tentar conhecer de tudo, um pouco de cada coisa.
Você não sabe da onde vai encontrar sua paixão... Seja design, UX, CSS, Javascript, Arquitetura, Banco, Infra, Redes. Pode ser que você goste de várias coisas e seja realmente bom com algumas delas.

**Comment by eduardojmatos** [comment]
@furabio acho que nem faz muito sentido debatermos esse post mano... Tem tanta coisa errada escrita nessa resposta que não vale perdermos tempo. É muita falta de conhecer o mercado, de ter experiência com front-end ou de preconceito com a área que acredito que não devamos gastar tempo com essa história 😞

**Comment by felipefialho** [comment]
Acabei vendo por cima, esse link que o @woliveiras postou no sábado e não quis estragar meu fim de semana com afirmações tao absurdas.

Infelizmente o mercado esta cheia de gente assim, extremas e que não conseguem compreender o que é a web.

Imagina esse sujeito, que diz ser possivel virar especialista CSS e HTML em 4 dias desenvolvendo uma aplicação escalavel, com suporte a vários navegadores e milhões de dispositivos...

Mas sinceramente, Darwin cuida disso. E podem ter certeza que ele não conseguiria vaga em nenhuma empresa que pessoas presentes nesse forum trabalham.

**Comment by Marcelosilva10** [comment]
@LFeh o cara falar que é especialista em 4 dias está de brincadeira, isso da para ter uma ideia o tipo de profissional que se tornou, o sujeito.
Não existem segurança, performasse, mobile praticamente toda a arquitetura do site para ele?

**Comment by bernardodiasc** [comment]
Outro dia um amigo me mostrou o site pessoal dele, digo, sobre o profissional de web. No site haviam barras de progresso de cada linguagem e uma pontuação que ele atribuía a si mesmo.

Minha indagação foi: como ele podia se definir nível em 9 ou 7 ou 3 em qualquer coisa?

Se ele se diz 9 em HTML numa escala de 0 a 10, significa que ele escreve drafts e fixa bugs no core dos browsers? Se fosse 10 então ele saberia absolutamente tudo que há pra saber? Como ele saberia que sabe tudo com 2 anos de experiência em webdev?

Fica ai a reflexão.

**Comment by lagden** [comment]
as vezes ele é um excelente faixa branca!!!

**Comment by rubenmarcus** [comment]
sim mas no mercado num geral falando em valores ($$$) DBA e Back-End > Front Engineer > Front Designer (UI/UX) > Webdesigner

Design é um mercado que paga muito mal pro pessoal em geral. Eu já trabalhei em agências com estagiários que faziam P&P em ESPM, FAAP, Belas Artes, PUC, e além de layout,diagramação etc, os caras tinham que codar telas, e os caras ganhavam algo em torno de 1500, e mal sabiam o que faziam, apesar de aprender rápido.

Da mesma forma vi, e ainda vejo programadores, que tem cabeça de programador, fizeram Analise de Sistemas ou não, ou que são auto ditada, e tem conhecimento em AI e etc, e ganha menos de 5mil.

A minha opinião é ..

tem que se pensar no profissional com certeza, ter uma boa carreira, currículo e renda que permita uma qualidade de vida boa.

Mas o pessoal tambem é muito importante.
Você é do tipo que fuça blogs de tecnologia,fóruns, ta sempre colocando em experimentação coisas novas, e se desafiando?
Programa e faz do seu trabalho render que nem vê as horas passar?
Está pronto pra lidar com pepinos, projetos e chefes que não vão entender bulhufas do que você faz e vão falar absurdos, que você vai ser obrigado a lidar?
Tem vontade de fazer projetos opensource, criativos, projetos para enriquecer a comunidade e a base de usuários em geral?

Isso seja no design,front ou back..

Tem as referências de profissionais bem sucedidos? Se ve fazendo o mesmo ou melhor?

Acho que são essas perguntas que eu me faço todos os dias, que todo profissional devia fazer

Desmerecer outra função por ser "menos importante" é um erro grotesco, porque o "importante" tudo depende do que e do momento.

Essa é minha opinião e um pouco do que vejo no mercado..

Vejo muito profissional com 30 siglas, 15 certificados, um monte de frameworks e conceitos, e na hora de por em pratica, nunca colocou um projeto em produção..

tem isso tambem.

Tem que ter a vontade de entregar e resolver as coisas.. Sempre da melhor forma, mas tambem sempre ir atrás.. esse é um aspecto importante..

Eu acho que pra qualquer front, aprender acessar um banco de dados não é algo de outro mundo..  Esse post foi totalmente viajado.. 

E num geral pelo que eu vejo, pra quem tem lógica e boa experiência ,novas ferramentas são algo que se aprende rápido e geralmente não são um obstáculo.

**Comment by rubenmarcus** [comment]
@bernardodiasc  Eu tava vendo naquele site upwork.com existe um certo "quiz" de conhecimentos, que te avalia e ajuda na hora de pegar projetos.

Mas ao meu ver eu acho esses portfolios um saco.. O meu mesmo eu coloquei pelo tempo de experiência com as ferramentas e linguagens e projetos que já apliquei. Acho essa uma forma mais honesta e menos vaga.

Apesar que eu entenda que esses profissionais coloque por exemplo 10 em HTML , ele dizendo que faria tudo que o contratante pedir se tratando de HTML,CSS ou JS..
